src/reconcile/rec_visibility.c

```diff
@@ -745,7 +746,7 @@ __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_UPDATE *first_up
              */
             if (F_ISSET(r, WT_REC_CHECKPOINT) &&
               (!F_ISSET(conn, WT_CONN_PRESERVE_PREPARED) ||
-                upd->upd_start_ts > conn->txn_global.checkpoint_timestamp)) {
+                upd->prepare_ts > conn->txn_global.checkpoint_timestamp)) {
```

Comment:
The max_ts variable in this function needs to consider prepare_ts

Reply:
edited